tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js

```diff
@@ -44,6 +44,85 @@
 			helpers.restoreStub(ns.support.shape, "circle");
 		});
 
+		QUnit.test("_init", 4, function (assert) {
+			var element = document.getElementById("scrollview-page-w"),
+				scrollview = new Scrollview();
+
+			helpers.stub(Scrollview.prototype, "_setBouncingEffect", function () {
+				assert.ok(true, "_setBouncingEffect should be called");
+			});
+
+			scrollview.element = element;
+			scrollview._init();
+
+			assert.equal(scrollview.maxScrollX, 0, "maxScrollX is 0");
+			assert.equal(scrollview.maxScrollY, 0, "maxScrollY is 0");
+			assert.notEqual(scrollview.bouncingEffect, undefined, "bouncing effect is created");
+
+			helpers.restoreStub(Scrollview.prototype, "_setBouncingEffect");
+		});
+
+		QUnit.test("_start & _end", 6, function (assert) {
+			var element = document.getElementById("scrollview-page-w"),
+				scrollview = new Scrollview();
+
+			helpers.stub(Scrollview.prototype, "_setBouncingEffect", function () {
+				assert.ok(true, "_setBouncingEffect should be called");
+			});
+
+			scrollview.element = element;
+			scrollview._build(element);
+			scrollview._init();
+
+			scrollview._start();
+
+			assert.equal(scrollview.scrolled, false, "scrolled is false");
+			assert.equal(scrollview.dragging, true, "dragging is true");
+			assert.equal(scrollview.scrollCanceled, false, "scrollCanceled is false");
+
+			helpers.stub(scrollview.bouncingEffect, "dragEnd", function () {
+				assert.ok(true, "dragEnd should be called");
+			});
+
+			scrollview._end();
+
+			assert.equal(scrollview.dragging, false, "dragging is false");
+
+			helpers.restoreStub(scrollview.bouncingEffect, "dragEnd");
+			helpers.restoreStub(Scrollview.prototype, "_setBouncingEffect");
+		});
+
+		QUnit.test("showBouncingEffect & hideBouncingEffect", 5, function (assert) {
+			var element = document.getElementById("scrollview-page-w"),
+				scrollview = new Scrollview();
+
+			helpers.stub(Scrollview.prototype, "_setBouncingEffect", function () {
+				assert.ok(true, "_setBouncingEffect should be called");
+			});
+
+			scrollview.element = element;
```

Comment:
these 3 lines seems not needed.

You need stub for scrollview.scroller: this.scroller.getBoundingClientRect().height

Reply:
If I remove the 3 lines, I got following message.
```
TypeError: Can not read property 'drag' of undefined
```
If there is no process for initializing the widget, fail to get the bouncingEffect object.

Reply:
Hi :),
Implementation of `showBouncingEffect` method is simple.
```
			prototype.showBouncingEffect = function (place) {
				var bouncingEffect = this.bouncingEffect;

				if (place === "end") {
					bouncingEffect.drag(0, -this.scroller.getBoundingClientRect().height);
				} else {
					bouncingEffect.drag(0, 0);
				}
			};
```
This is unit test. Its purpose to test only this method.

You need two mocks.
one:
```
scrollview.bouncingEffect = {
   drag: function (arg1, arg2) {
       assert.ok(true, "Method bouncingEffect has been called");
       assert.equal(arg1, 0, "Method bouncingEffect: first argument equals 0");
   }
}
```
two:
```
scrollview.scroller = {
    getBoundingClientRect: function () {
         assert.ok(true, "Method getBoundingClientRect has been called");
         return {
              height: 100; // a sample value;
         }
    }
}
```

Reply:
I see. I added your suggestion.